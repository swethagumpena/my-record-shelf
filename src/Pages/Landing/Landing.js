import React from 'react';
import { useHistory } from 'react-router-dom';
import styles from './Landing.module.scss';

const Landing = () => {
  const history = useHistory();

  return (
    <div className={styles.page}>
      <div className={styles.container}>
        <p>:((</p>
        <p>Seems a bit empty in here</p>
        <button type="button" onClick={() => history.push('/songs')}>sync</button>
      </div>

    </div>
  );
};
export default Landing;
