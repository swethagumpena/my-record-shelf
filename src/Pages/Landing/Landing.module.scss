@import "../../styles/styles.scss";

.page {
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    // background-image: linear-gradient(#ADD2FF, white);
}

.container{
    display: flex;
    flex-direction: column;
    align-items: center;
}

// .container h3 {
//     font-family: "Inconsolata", monospace;
//     font-size: 4.5rem;
//     letter-spacing: 0.25vw;
//     margin: 1rem;
//     color: $primary-text-color;
// }
 
.container p {
    font-family: 'Poppins', sans-serif;
    font-size: 1rem;
    padding: 0.5rem;
}

.container button {
    font-family: 'Poppins', sans-serif;
    padding: 1rem;
    border-radius: 2rem;
    padding: 0.5rem 2rem;
    color: white;
    background-color: #4C63AA;
    font-weight: bold;
    margin: 0.5rem;
}

@include media-mixin($mobile-breakpoint){
    .container h3{
        font-size: 3rem;
    }
    .container p{
        font-size: 1.5rem;
    }
    .container button{
        padding: 1rem;
    }
}
