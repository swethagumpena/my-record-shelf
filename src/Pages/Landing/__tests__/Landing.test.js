import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Landing from '../Landing';

describe(Landing.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<Landing />);
    expect(container).toMatchSnapshot();
  });

  test('should display intro content', () => {
    render(<Landing />);
    expect(screen.getByText(':(('));
    expect(screen.getByText('Seems a bit empty in here'));
  });

  test('should display a sync  started button', () => {
    render(<BrowserRouter><Landing /></BrowserRouter>);
    const gettingStartedElement = screen.getByText('sync');
    expect(gettingStartedElement.tagName).toBe('BUTTON');
    fireEvent.click(gettingStartedElement);
    expect(document.location.href).toBe('http://localhost/songs');
  });
});
