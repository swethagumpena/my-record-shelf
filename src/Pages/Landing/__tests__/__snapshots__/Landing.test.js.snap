// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match snapshot 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="container"
    >
      <p>
        :((
      </p>
      <p>
        Seems a bit empty in here
      </p>
      <button
        type="button"
      >
        sync
      </button>
    </div>
  </div>
</div>
`;
